src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -151,6 +156,12 @@ class DatasetFilter:
     task_categories: Optional[Union[str, List[str]]] = None
     task_ids: Optional[Union[str, List[str]]] = None
 
+    def __post_init__(self):
+        warnings.warn(
+            "DatasetFilter is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to the `list_datasets` method.",
```

Comment:
To make it work with `@expect_deprecation` in the tests

```suggestion
            "'DatasetFilter' is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to the `list_datasets` method.",
```